tests/Unit/Queue/Actions/RunOrQueueJobActionTest.php

```diff
@@ -0,0 +1,330 @@
+                    actual: self
+                                    ::makeAction(
+                                        expectedDispatchJob: true,
+                                        expectedRunJob: false,
+                                        expectedCommand: null,
+                                        job: $job,
+                                    )
+                                        ->execute(job: $job, shouldQueue: true),
+                ),
+            ],
+            '$shouldQueue=false forces the job tu run' => [
+                static fn () => Assert::assertEquals(
+                    expected: 'Test',
+                    actual: self
+                                    ::makeAction(
+                                        expectedDispatchJob: false,
+                                        expectedRunJob: true,
+                                        expectedCommand: null,
+                                        job: $job,
+                                    )
+                                        ->execute(job: $job, shouldQueue: false),
+                ),
+            ],
+            '$shouldQueue=false forces the job tu run, setupBeforeRun called' => [static function () use ($job) {
+                $setupBeforeRunCalled = false;
+                Assert::assertEquals(
+                    expected: 'Test',
+                    actual: self
+                        ::makeAction(
+                            expectedDispatchJob: false,
+                            expectedRunJob: true,
+                            expectedCommand: null,
+                            job: $job,
+                        )
+                            ->execute(
+                                job: $job,
+                                setupBeforeRun: static function (Job $givenJob) use ($job, &$setupBeforeRunCalled) {
+                                    Assert::assertSame($job, $givenJob, 'Job should be same');
+                                    $setupBeforeRunCalled = true;
+                                },
+                                shouldQueue: false
+                            ),
+                );
+                Assert::assertTrue($setupBeforeRunCalled, 'setupBeforeRun should be trigered');
+            }],
+        ];
+    }
+
+    /**
+     * @param Closure():void $assert
+     *
+     * @dataProvider dataNoCommand
+     */
+    public function testNoCommand(Closure $assert): void
+    {
+        $assert();
+    }
+
+    /**
+     * @return array<string|int, array{0: Closure(static):void}>
+     */
+    public function dataWithCommand(): array
+    {
+        $job = new WithoutCommandJob('Test');
+        $command = $this->makeCommand([]);
+
+        return [
+            'queue not set, runs the job' => [
+                static fn () => Assert::assertEquals(
+                    expected: 'Test',
+                    actual: self
+                                    ::makeAction(
```

Comment:
```
self
    ::makeAction()
```
je žádoucí?

Reply:
jsem si nevšiml, to mě musel ecs/rector zformátovat, původně to bylo na zarovnané pěkně s `->`, očividně to umí jenom když je `$this`

pokud bych měl pěkně $this tak to bude vypadat hezky takto se mi to nedaří hezky zarovnat.

Pohrál jsem si stím, není to co bych chtěl ale je to dostačující